OneSignalSDK/onesignal/in-app-messages/src/main/java/com/onesignal/inAppMessages/internal/InAppMessagesManager.kt

```diff
@@ -451,12 +451,19 @@ internal class InAppMessagesManager(
      *
      * Make all messages with redisplay available if:
      * - Already displayed
-     * - At least one Trigger has changed
+     * - At least one existing Trigger has changed OR a new trigger is added when there is only dynamic trigger
      */
-    private fun makeRedisplayMessagesAvailableWithTriggers(newTriggersKeys: Collection<String>) {
+    private fun makeRedisplayMessagesAvailableWithTriggers(
+        newTriggersKeys: Collection<String>,
+        isNewTriggerAdded: Boolean,
+    ) {
         for (message in messages) {
-            if (!message.isTriggerChanged && redisplayedInAppMessages.contains(message) &&
-                _triggerController.isTriggerOnMessage(message, newTriggersKeys)
+            if (!message.isTriggerChanged &&
+                redisplayedInAppMessages.contains(message) &&
+                (
+                    _triggerController.isTriggerOnMessage(message, newTriggersKeys) ||
+                        isNewTriggerAdded && _triggerController.messageHasOnlyDynamicTriggers(message)
```

Comment:
This `if` statement is now well over the line of what is considered readable. Let's split this up so it is more readable before we merge this.

Reply:
Agreed, the if statement is now broken into multiple descriptive boolean values to enhance readability.